### PR TITLE
Introduce initial openapi.yaml from server swagger

### LINKED ISSUE #61: Provide 2 TSP specifications for currently supported APIs and proposed APIs
Right now the TSP specification contains currently supported and used APIs as well as proposed APIs.
- It's not clear what is actually used and supported and what is a proposal.
- Split the TSP yaml description into 2 files:
> 1. Currently implemented and supported APIs
> 2. Future, proposed APIs (+ APIs of bullet 1)
- Check if https://github.com/swagger-api/swagger-core can be used to automatically generate 1's.

### PR DESCRIPTION
**Pull request by marco-miller**
Add the yaml file currently generated by the Swagger endpoint recently added to trace-server.
- This yaml file is the openapi or TSP that server currently implements, explicitly in the code.
- This yaml file differs from API.yaml which has been manually documented up until now.
- Hence, the hereby added openapi.yaml is the true current (reference) TSP, as opposed to API.yaml's purposely edited one.
- The heading commits of this PR come from using [these currently open changes](https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/186185) locally.

Add a corresponding README section that explains the full context, while listing steps to undertake and current ways of generating openapi.yaml.
- Also in README, remove duplicated links while listing them.
- Add license header to new yaml with script also in README; removes Jersey's WADL.

This change assumes a companion commit (#63) in the gh-pages branch, that adds a ./swagger/index.html file linking to this new openapi.yaml file. That added index file is likely a copy of ./index.html in gh-pages, only not pointing to API.yaml as such.

Fixes #61 partially as first steps.

Signed-off-by: Marco Miller <marco.miller@ericsson.com>

### CODE REVIEW

**Comment by marco-miller** [comment]
> I am not convinced by this approach, I would argue we could just update the current instead of having two APIs... if someone comes out of nowhere and sees two APIs... I suspect their reaction would be odd.
> 
> Also, we need to determine which one we want. Honestly, the one we have from trace compass is functional, but not clean, the one we have here is clean but not 100% functional. I assume there should be an in between. My 2c though.

Thanks @MatthewKhouzam for the review. @bhufmann, should we change our originally wanted approach based on the 2 APIs? I'm fine either way, but we need to decide on if we keep both or not, at least for the time being. I agree with @TheMatthew that publishing 2 APIs at once for a while might momentarily be confusing. But again, these changes are step-wise and incremental anyway.

File: README.md [hunk 1]
```diff
@@ -6,10 +6,57 @@ This protocol is built to decouple the backend and frontend of trace analysers,
...
+
+### Alternate TSP version
+
+Above, `API.yaml` has been the manually documented TSP version up until now.
+
+* That version is the one shown by the default [github pages][6].
+* That version does not fully match what trace-server currently supports; some endpoints differ or are missing in server.
+* Aligning the TSP documented that way with server's own endpoints is a work in progress.
+
+In the meantime, Swagger has recently been added to trace-server.
```
Comment (bhufmann) [inline README.md:37]: The Trace Compass trace server is an example implementation or reference implementation of the TSP server side. In general the TSP is independent to any specific client or server implementation. Could please clarify that and that you referring to the "Trace Compass trace server" here as a reference implementation which is used to refine the TSP?

**Comment by MatthewKhouzam** [comment]
> Thanks @MatthewKhouzam for the review. @bhufmann, should we change our originally wanted approach based on the 2 APIs? I'm fine either way, but we need to decide on if we keep both or not, at least for the time being. I agree with @TheMatthew that publishing 2 APIs at once for a while might momentarily be confusing. But again, these changes are step-wise and incremental anyway.

Any news here?

**Comment by marco-miller** [comment]
> Yeah a bit tricky. Maybe the one from Trace Compass swagger should be in a sub-directory for now, so tit's less confusing? And the index.html we don't publish then until it's more progressed?

That was the goal of #63 as this PR's very companion.

**Review by MatthewKhouzam — APPROVED** [review]
I am OK with this.

**Pull request merged**